craft_providers/lxd/lxd_provider.py

```diff
@@ -126,6 +126,26 @@ def launched_environment(
 
         :raises LXDError: if instance cannot be configured and launched.
         """
+        projects = self.lxc.project_list(remote=self.lxd_remote)
+        if self.lxd_project in projects:
+            devices = self.lxc.profile_show(
+                project=self.lxd_project, profile="default", remote=self.lxd_remote
+            ).get("devices")
+            if not devices:
+                # Project exists but the default profile is ill-formed, tell the user to
+                # delete the project and start over.
+                raise LXDError(
```

Comment:
Unfortunately deleting the broken projects ourselves involves deleting everything the project is using first, and this is too risky to implement without better investigation and careful testing.

Reply:
this you be a good scenario to have a test for, aside from that, I generally approve of the appoach

Reply:
I agree that this new check should have a test